proof.go

```diff
@@ -320,30 +322,86 @@ func (proof Proof) VerifyLeafHashes(nth *NmtHasher, verifyCompleteness bool, nID
 	// rightSubtrees only contains the subtrees after r.End
 	rightSubtrees := nodes
 
-	if verifyCompleteness {
-		// leftSubtrees contains the subtree roots upto [0, r.Start)
-		for _, subtree := range leftSubtrees {
-			leftSubTreeMax := MaxNamespace(subtree, nth.NamespaceSize())
-			if nID.LessOrEqual(namespace.ID(leftSubTreeMax)) {
-				return false, ErrFailedCompletenessCheck
-			}
+	// leftSubtrees contains the subtree roots upto [0, r.Start)
+	for _, subtree := range leftSubtrees {
+		leftSubTreeMax := MaxNamespace(subtree, nth.NamespaceSize())
+		if nID.LessOrEqual(namespace.ID(leftSubTreeMax)) {
+			return ErrFailedCompletenessCheck
 		}
-		for _, subtree := range rightSubtrees {
-			rightSubTreeMin := MinNamespace(subtree, nth.NamespaceSize())
-			if namespace.ID(rightSubTreeMin).LessOrEqual(nID) {
-				return false, ErrFailedCompletenessCheck
-			}
+	}
+	for _, subtree := range rightSubtrees {
+		rightSubTreeMin := MinNamespace(subtree, nth.NamespaceSize())
+		if namespace.ID(rightSubTreeMin).LessOrEqual(nID) {
+			return ErrFailedCompletenessCheck
 		}
 	}
+	return nil
+}
+
+// ComputeRootWithBasicValidation computes the Merkle root from a given proof and a set of leaf hashes,
+// performing basic validation steps prior to computing the root.
+//
+// If isNamespace is true, it additionally validates:
+//   - That the leaf hashes belong to the specified namespace.
+//   - That the proof is complete for the given namespace.
+//
+// Parameters:
+//   - nth: The NMT hasher instance used for validation and root computation.
+//   - nID: The namespace ID that the proof is expected to correspond to.
+//   - leafHashes: The hashes of the leaves being proven.
+//   - isNamespace: A flag indicating whether namespace-specific validation should be performed.
+//
+// Returns:
+//   - The computed root hash if all checks pass.
+//   - An error if any validation fails or root computation fails.
+func (proof Proof) ComputeRootWithBasicValidation(nth *NmtHasher, nID namespace.ID, leafHashes [][]byte, isNamespace bool) ([]byte, error) {
```

Comment:
[blocking] this is a new exported function and it's lacking unit tests.

Reply:
it doesn't add any new logic. Just a wrapper function using existing functions. Also, more tests will be covered as part of https://github.com/celestiaorg/nmt/issues/292